core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -92,7 +93,16 @@ public TableScan appendsAfter(long fromSnapshotId) {
     protected CloseableIterable<FileScanTask> planFiles(TableOperations ops, Snapshot snapshot, Expression rowFilter,
                                                         boolean ignoreResiduals, boolean caseSensitive,
                                                         boolean colStats) {
-      CloseableIterable<ManifestFile> filtered = filterManifests(manifests(), rowFilter, caseSensitive);
+      Map<Integer, PartitionSpec> specsById = table().specs();
```

Comment:
In a few other places, we copy specs into a new map via `Maps.newHashMap` to make sure Java collections are used. This is done to avoid any potential Kryo issues as `ScanTasks` are serializable. Maybe, we should do the same here.

At the end of the method, we call `ops.io(), ops.current().specsById()` to construct a task. In theory, we could pass `Table` object to `ManifestReadTask` and then access `FileIO` via `table.io()` and copy specs into a Java map in the constructor. Up to you. At least, no need to call `ops.current().specsById()` as we already have specs.

Reply:
Hm, not sure I completely see, this specsById is just used to lookup a cache right, before the ManifestReadTask is returned?  

So I didn't change this line but changed the ManifestReadTask constructor to take table and there do Maps.newHashMap, let me know if thats what you wanted

Reply:
This looks good to me.